src/main/webapp/js/mpData.js

```diff
@@ -8,6 +8,28 @@
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
+function displayLibertyVersion() {
+    getRuntimeRequest();
+}
+
+function getRuntimeRequest() {
+    var url = location.origin + "/system/runtime";
+    var req = new XMLHttpRequest();
+    var table = document.getElementById("systemPropertiesTable");
+
+    req.onreadystatechange = function () {
+        if (req.status === 200) {
+            var version = req.responseText;
+            if (version != "") {
+                var appTitle = document.getElementById("appTitle");
+                appTitle.innerText = "Open Liberty " + version + " System Properties Sample";
```

Comment:
Thinking about XSS in the `.innerText` call, is the value of `version` from the MBean `WebSphere:feature=kernel,name=ServerInfo` user controllable?  It looks like OWASP [documents](https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html#usually-safe-methods) that `.innerText` is usually safe with some edge cases.

> Some papers or guides advocate its use as an alternative to innerHTML to mitigate against XSS in innerHTML. However, depending on the tag which innerText is applied, code can be executed.

Reply:
it is not, the mbean returns the liberty version which is provided by the Liberty kernel (and if someone can muck with that, XSS is the least of their worries).